doc/howto/many-executables.rst

```diff
@@ -0,0 +1,150 @@
+argument after the executable name:
+
+.. code:: shell-session
+
+  $ dune init exe hello intro_ocaml
+  Success: initialized executable component named hello
+
+  $ dune init exe fibonacci intro_ocaml
+  Success: initialized executable component named fibonacci
+
+  $ dune init exe watchfs intro_lwt --libs=lwt.unix
+  Success: initialized executable component named watchfs
+
+  $ dune init exe verilog intro_hardcaml --libs=base,hardcaml --ppx=ppx_jane,ppx_hardcaml
+  Success: initialized executable component named verilog
+
+Folder Structure
+----------------
+
+.. code:: shell-session
+
+  $ dune clean
+  $ tree
+  .
+  ├── dune-project
+  ├── intro_hardcaml
+  │   ├── dune
+  │   └── verilog.ml
+  ├── intro_lwt
+  │   ├── dune
+  │   └── watchfs.ml
+  └── intro_ocaml
+      ├── dune
+      ├── fibonacci.ml
+      └── hello.ml
+
+You can execute these by including the path:
+
+.. code:: shell-session
+
+  $ dune exec intro_ocaml/hello.exe
+  Hello, World!
+
+Adding Libraries
+----------------
+
+As you begin to share modules between executables, you can add them individually:
+
+.. code:: shell-session
+
+  $ dune init lib greet lib/util
+  Success: initialized library component named greet
+
+  $ echo 'let hi w = "Hi " ^ w' > lib/util/greet.ml
+
+Or you can, as with the default ``dune init proj`` layout, add a directory of submodules:
+
+.. code:: shell-session
+
+  $ dune init lib intro_ocaml lib/intro_ocaml
+  Success: initialized library component named intro_ocaml
+
+  $ echo 'let n = 0xF00F' > lib/intro_ocaml/magic.ml
+  $ echo 'let bump = (+) 1' > lib/intro_ocaml/math.ml
+
+Which you can use like normal, adding the named libraries to an executable's stanza:
+
+.. code:: dune
+
+  ; from intro_ocaml/dune
+  (executable
+   (name hello)
+   (libraries greet intro_ocaml)) ; new
+
+.. code:: ocaml
+
+  (* new contents of intro_ocaml/hello.ml *)
+  let () =
+    print_endline (Greet.hi "world");
+    print_int Intro_ocaml.(Math.bump Magic.n);
+    print_newline ()
+
+.. code:: shell-session
+
+  $ dune exec intro_ocaml/hello.exe
+  Hi world
+  61456
+
+Adding Public Names
+-------------------
+
+Now, suppose some of your many small executables start to stand out from the
+crowd, like that excellent ``intro_ocaml/hello.ml``. To give it a proper name
+and make it installable, you first need a package.
+
+If you created the dune project with ``dune init proj``, you were already given
+a package. If not, here is a minimal addition to your ``dune-project``:
+
+.. code:: dune
+
+  (package
+   (name tutorial)
+   (synopsis "Some tutorial programs")
+   (depends ocaml))
+
+You can then update the executable's stanza:
+
+.. code:: dune
+
+  (executable
+   (name hello)
+   (public_name hello) ; new
+   (libraries greet intro_ocaml))
+
+And execute it:
+
+.. code:: shell-session
+
+  $ dune exec hello
+  Hi world
+  61456
+
+Or install it and execute it:
+
+.. code:: shell-session
+
+  $ dune build
+  $ dune install
```

Comment:
I would strongly recommend against using `dune install` as that messes with potential switches, the system etc.